fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/FabricModel.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1;
+
+import net.minecraft.data.client.Model;
+
+import net.fabricmc.fabric.api.datagen.v1.builder.DisplayBuilder;
+import net.fabricmc.fabric.api.datagen.v1.builder.ElementBuilder;
+import net.fabricmc.fabric.api.datagen.v1.builder.OverrideBuilder;
+
+/**
+ * Fabric-provided extensions for {@link Model}.
+ *
+ * <p>Note: This interface is automatically implemented on all models via Mixin and interface injection.
+ */
+public interface FabricModel {
+	/**
+	 * Adds an entry to the <code>elements</code> property of a model.
+	 * Element entries consist of a pair of opposite vertices of a cuboid to draw the element out as, with an optional
+	 * rotation, shading and set of rendered faces.
+	 *
+	 * @param position A {@link DisplayBuilder.Position} to set the property for. Can be either hand of either first or
+	 *                    third person, the head, the ground (as a dropped item), within GUIs or within item frames
+	 *                    (<code>FIXED</code>).
+	 * @param builder A {@link DisplayBuilder} to build the required display property from.
+	 * @return The current newly-modified {@link Model} instance.
+	 */
+	default Model withDisplay(DisplayBuilder.Position position, DisplayBuilder builder) {
```

Comment:
This should throw instead. See for example https://github.com/FabricMC/fabric/blob/1.19.3/fabric-loot-api-v2/src/main/java/net/fabricmc/fabric/api/loot/v2/FabricLootTableBuilder.java.